azurerm/internal/services/apimanagement/api_management_diagnostic_resource.go

```diff
@@ -95,6 +144,70 @@ func resourceApiManagementDiagnosticCreateUpdate(d *schema.ResourceData, meta in
 		},
 	}
 
+	if samplingPercentage, ok := d.GetOk("sampling_percentage"); ok {
+		parameters.Sampling = &apimanagement.SamplingSettings{
+			SamplingType: apimanagement.Fixed,
+			Percentage:   utils.Float(samplingPercentage.(float64)),
+		}
+	} else {
+		parameters.Sampling = nil
+	}
+
+	if alwaysLogErrors, ok := d.GetOk("always_log_errors"); ok && alwaysLogErrors.(bool) {
+		parameters.AlwaysLog = apimanagement.AllErrors
+	}
+
+	if verbosity, ok := d.GetOk("verbosity"); ok {
+		switch verbosity.(string) {
+		case string(apimanagement.Verbose):
+			parameters.Verbosity = apimanagement.Verbose
+		case string(apimanagement.Information):
+			parameters.Verbosity = apimanagement.Information
+		case string(apimanagement.Error):
+			parameters.Verbosity = apimanagement.Error
+		}
+	}
+
+	logClientIP, _ := d.GetOk("log_client_ip")
+	if logClientIP != nil {
+		parameters.LogClientIP = utils.Bool(logClientIP.(bool))
+	}
+
+	if httpCorrelationProtocol, ok := d.GetOk("http_correlation_protocol"); ok {
+		switch httpCorrelationProtocol.(string) {
+		case string(apimanagement.HTTPCorrelationProtocolNone):
+			parameters.HTTPCorrelationProtocol = apimanagement.HTTPCorrelationProtocolNone
+		case string(apimanagement.HTTPCorrelationProtocolLegacy):
+			parameters.HTTPCorrelationProtocol = apimanagement.HTTPCorrelationProtocolLegacy
+		case string(apimanagement.HTTPCorrelationProtocolW3C):
+			parameters.HTTPCorrelationProtocol = apimanagement.HTTPCorrelationProtocolW3C
+		}
```

Comment:
and do the same conversion here instead of a switch

Reply:
Updated, also updated `api_management_api_diagnostic.go`